File: WindowsRegular Expressions/4975sourcecode/chapter5/5-9/recipe5-9.vb
Imports System
Imports System.IO
Imports System.Text.RegularExpressions
Public Class Recipe

    Private Shared _Regex As Regex = _
    	New Regex("<script language=""javascript"">([^<""]*|[^<]*<[^/][^<]*|(([^<""]|[^<]*<[^/][^<])*""[^""]*""([^<""]|[^<]*<[^/][^<])*)*)?</script>", _
    		RegexOptions.IgnoreCase Or RegexOptions.Multiline)

    Public Sub Run(ByVal fileName As String)
        Dim line As String
        Dim sr As StreamReader = File.OpenText(fileName)
        line = sr.ReadToEnd()
        If Not line Is Nothing
            For Each m As Match In _Regex.Matches(line)
            	Console.WriteLine("Found match '{0}'", m.ToString())
            Next
        End If
        sr.Close()
    End Sub

    Public Shared Sub Main(ByVal args As String())
        Dim r As Recipe = New Recipe
        r.Run(args(0))
    End Sub

End Class

File: WindowsRegular Expressions/4975sourcecode/chapter5/5-9/recipe5-9.cs
using System;
using System.IO;
using System.Text.RegularExpressions;

public class Recipe
{

	private static Regex _Regex = 
		new Regex(@"<script language=""javascript"">([^<""]*|[^<]*<[^/][^<]*|(([^<""]|[^<]*<[^/][^<])*""[^""]*""([^<""]|[^<]*<[^/][^<])*)*)?</script>", 
		          RegexOptions.IgnoreCase | RegexOptions.Multiline);
	
	public void Run(string fileName)
	{
		String line;
		using (StreamReader sr = new StreamReader(fileName))
		{
			line = sr.ReadToEnd();
			foreach (Match m in _Regex.Matches(line))
			{
				Console.WriteLine("Found match '{0}'", 
					m.ToString());
			}
		}
	}	

	public static void Main( string[] args )
	{
		Recipe r = new Recipe();
		r.Run(args[0]);
	}
}

